sycl/include/sycl/ext/oneapi/backend/level_zero.hpp

```diff
@@ -40,6 +39,10 @@
 
 namespace sycl {
 inline namespace _V1 {
+
+template <bundle_state State>
+class kernel_bundle;
```

Comment:
As I was working on #16030 and putting even more forward-declarations everywhere, I though that having a dedicated version of every header which only provides forward declarations would actually be cool.

Because some classes are templated, their forward-declaration may not be trivial and it may require more forward-declarations. Having a ready-to-use "canonical" forward-declaration in form of a separate header would shorten the codebase and make it simpler (i.e. that would just be a `#include` list without any extra lines of code).

There is a thing that some headers provide multiple classes and functions and therefore we could "import" more forward-declarations than we need, but I assume that it would be cheaper anyway that including the original header, so I don't think that it is a strong enough point against the idea.